openforcefield/typing/engines/smirnoff/parameters.py

```diff
@@ -2811,18 +3176,21 @@ def _assign_fractional_bond_orders(
                 match.reference_molecule.assign_fractional_bond_orders(
                     toolkit_registry=toolkit_registry,
                     use_conformers=match.reference_molecule.conformers,
+                    bond_order_model=self.fractional_bondorder_method.lower()
                 )
 
             # scale k based on the bondorder of the central bond
             if self.fractional_bondorder_interpolation == "linear":
                 # we only interpolate on k
-                k = self._linear_interpolate_k(
+                k = _linear_inter_or_extrapolate(
                     k_bondorder, central_bond.fractional_bond_order
                 )
             else:
-                raise Exception(
-                    "Fractional bondorder treatment {} is not implemented.".format(
-                        self.fractional_bondorder_method
+                # TODO: This code is effectively unreachable due to the the _allow_only converter used in this
+                #       ParameterAttribute's definition, which only allows "linear". Remove?
+                raise FractionalBondOrderInterpolationMethodUnsupportedError(
+                    "Fractional bondorder interpolation method {} is not implemented.".format(
+                        self.fractional_bondorder_interpolation
```

Comment:
I'm fine with this code in here as an extra safeguard against bad behavior.

Reply:
Thanks. I'm wondering if it's a bad pattern to validate things multiple times like this (since then, if we wanted to change the behavior in the future, we'd have to track down _all_ the separate instances of validation). But I'll leave this in and we can learn what this pattern leads to.